### PR TITLE
break circular dependency: random/sync -> util -> random/sync

### PR DESCRIPTION
**Pull request by ken2812221**
LogPrintf  has acutally been moved to logging.h

### CODE REVIEW

**Comment by practicalswift** [comment]
Concept ACK

Let's get rid of circular dependencies! 👍

**Comment by MarcoFalke** [comment]
utACK 695951e3f853cfb1d7b22ddf409d9022993151a4

**Comment by promag** [comment]
utACK 695951e.

**Comment by practicalswift** [comment]
utACK 695951e3f853cfb1d7b22ddf409d9022993151a4

File: src/random.cpp [hunk 1]
```diff
@@ -11,15 +11,16 @@
 #include <compat.h> // for Windows API
 #include <wincrypt.h>
 #endif
-#include <util.h>             // for LogPrint()
-#include <utilstrencodings.h> // for GetTime()
+#include <logging.h>          // for LogPrint()
+#include <utiltime.h>         // for GetTime()
```
Comment (Empact) [inline src/random.cpp]: nit: unnecessary whitespace

File: src/random.cpp [hunk 2]
```diff
@@ -11,15 +11,16 @@
...
 
 #include <stdlib.h>
 #include <chrono>
 #include <thread>
 
 #ifndef WIN32
 #include <sys/time.h>
+#include <fcntl.h>
```
Comment (Empact) [inline src/random.cpp]: nit: sorting

File: src/sync.cpp [hunk 3]
```diff
@@ -4,9 +4,10 @@
 
 #include <sync.h>
 
+#include <map>
 #include <memory>
 #include <set>
-#include <util.h>
+#include <logging.h>
```
Comment (Empact) [inline src/sync.cpp]: nit: move the stdlib includes below (as is I believe clang-format will intermix them in the sort)

**Comment by Empact** [comment]
Just nits I noticed on second look, fine to include with.

File: src/sync.cpp [hunk 4]
```diff
@@ -4,13 +4,15 @@
 
 #include <sync.h>
 
-#include <memory>
 #include <set>
```
Comment (Empact) [inline src/sync.cpp]: One more :P

**Comment by Empact** [comment]
re-utACK 84f4194 - whitespace and include ordering changes only
